### DISCUSSION TITLE
Door Status Unknown

### DISCUSSION THREAD

**Discussion by 14yannick** [14yannick/HCPBridgeMqtt#2] [Ideas]
See https://github.com/pboschi/HCPBridgeMqtt/issues/3
When door is in position stopped then status UNKNOWN is returned. Do we should change it to "Partially Open" if door position is greater than 0. I have already add some more Doorstate given my the Motor but if we send a position or a Stop command then motor return status stopped who will give an UNKNOWN Status in MQTT.
@stomko11

**Comment by stomko11** [comment]
Well unknown is just the name. Basically you can do those checks outside of code - in HA for instance.

**Reply by 14yannick** [reply]
Good point. The motor publish several status.
Some where already in the code
- DOOR_OPEN_POSITION =  0x20,    
- DOOR_CLOSE_POSITION = 0x40,
- DOOR_HALF_POSITION =  0x80,
- DOOR_MOVE_CLOSEPOSITION = 0x02,
- DOOR_MOVE_OPENPOSITION = 0x01,
Based on the Analysis of thomas https://blog.dupas.be/posts/hoermann-uap-hcp1/ I added 2 State more
- DOOR_VENT_POSITION = 0x0A,
- DOOR_STOPPED = 0x00
Then I found out that two more can be observed when door is moving to the half opening and venting position.
- DOOR_MOVE_VENTPOSITION = 0x09,
- DOOR_MOVE_HALFPOSITION = 0x05,

There are mapped as following to an MQTT Status
- DOOR_OPEN_POSITION --> open
- DOOR_CLOSE_POSITION  --> closed
- DOOR_HALF_POSITION --> half
- DOOR_MOVE_CLOSEPOSITION --> closing
- DOOR_MOVE_OPENPOSITION --> opening
- DOOR_MOVE_VENTPOSITION --> opening
- DOOR_MOVE_HALFPOSITION --> opening
- DOOR_VENT_POSITION --> vent
- else  "UNKNOWN";

The status Stopped ist not mapped therfore it get Status unknown each time whe stop the door or set an opening percentage.
I think this status are then used to inform in which state the cover are. An based on the MQTT cover only 4 status are known there. state_open, state_opening, state_closed or state_closing.

Maybe two door Status are needed 

1. A Motor Status sending the status as they come from the hörmann device.
2. A door State with some logic where we set only the four Status expected from the MQTT Cover.

**Reply by stomko11** [reply]
Are you able to actually determine why this code from pboschi is misbehaving for me? All works fine, except when i restart ESP, door triggers automatically (opens if closed, closes if opened). I assume it may be due some extra checks being or not being present at boot. I moved to other fork that is not doing it, but it also does not have position slider (can't use cover.set_position service)

**Reply by 14yannick** [reply]
This behavior is quite strange have you tried to shut down your wifi when you restart your esp. To be sure that this misbehaving came not from an mqtt response to the mqtt discovery who send a close command (a close command open the door if it door is closed).  I have such routine on my lights ( turn off after power power came back after power failure)
